lib/client-assets.php

```diff
@@ -440,39 +440,39 @@ function gutenberg_register_vendor_script( $handle, $src, $deps = array() ) {
  */
 function gutenberg_extend_wp_api_backbone_client() {
 	// Post Types Mapping.
-	$post_type_rest_base_mapping = array();
-	foreach ( get_post_types( array(), 'objects' ) as $post_type_object ) {
-		$rest_base = ! empty( $post_type_object->rest_base ) ? $post_type_object->rest_base : $post_type_object->name;
-		$post_type_rest_base_mapping[ $post_type_object->name ] = $rest_base;
-	}
+	$post_type_rest_base_mapping               = array();
+	$post_type_rest_base_mapping['attachment'] = 'media';
```

Comment:
Why was this changed from the previous implementation where we use the post type objects to determine the `rest_base`? I don't think we want to hard-code attachment here, especially since we should respect the `rest_base` of any post type, without requiring that the plugin author opt in to `post_type_rest_base_mapping` (questioning whether the filter is even necessary, or redundant with `rest_base`).

Reply:
The original implementation changed the mappings for post => posts, and page => pages both where the model object is not plural. I think the rest base and route would be specified in the model, so I don't think we need it for determining the model.

Reply:
>The original implementation changed the mappings for post => posts, and page => pages both where the model object is not plural.

But this is handled in the client. Even in your current implementation you're passing the mapping in as the plural.

Handled here:

https://github.com/WordPress/wordpress-develop/blob/1c2b0762c7f1a91394c633ae812505919bd297ed/src/wp-includes/js/wp-api.js#L1190-L1223

I'm not really sure why such a mapping exists, but I don't know that we need to worry so much about singular-ifying custom types?

Even if we were to create a mapping, we'd probably want to take advantage of the existing `wpApiSettings.mapping` rather than creating a separate variable.

cc @adamsilverstein